pkg/controller/signer/signer-controller.go

```diff
@@ -88,6 +89,8 @@ func (r *ReconcileCSR) Reconcile(ctx context.Context, request reconcile.Request)
 	if err != nil {
 		if apierrors.IsNotFound(err) {
 			// Request object not found, could have been deleted after reconcile request.
+			// restore network status when CSR is deleted.
+			r.status.SetNotDegraded(statusmanager.CertificateSigner)
```

Comment:
What if there is more than one CSR that is failing and one of those gets removed?

Reply:
yes @kyrtapz, the status gets updated again when next signing attempt happens for another CSR.

Reply:
To summarize, I don't think we should change the CNO degraded state based on any particular CSR object, this should be reserved only for internal controller failures.